### PR TITLE
Update Developer Guide - Non Functional Requirements

### LINKED ISSUE #9: Update Developer Guide - Non Functional Requirements
[More details](https://nus-cs2103-ay2324s2.github.io/website/schedule/week7/project.html#4-update-the-dg-user-stories-glossary-nfrs-use-cases)

### PR DESCRIPTION
**Pull request by RyanNgWH**
Update developer guide to include non functional requirements for ClickConnect.

Closes #9.

### CODE REVIEW

File: docs/DeveloperGuide.md
Comment (ruijietay) [inline docs/DeveloperGuide.md:395]:
```suggestion
6. Should have a shallow learning curve. New users (including the non-technically savvy) should be able to pick up and start using the application efficiently within a week.
```

Should `non technically` be hyphenated instead? *(i.e. `non-technically`)*

**Review by ruijietay — COMMENTED** [review]
Just a small typo, but apart from that, LGTM!

File: docs/DeveloperGuide.md
Comment (RyanNgWH) [reply docs/DeveloperGuide.md]: You are right. Good catch!

File: docs/DeveloperGuide.md
Comment (dillontkh) [inline docs/DeveloperGuide.md:394]: I think the list numbering might have been messed up in the process, but otherwise the content looks good!

**Review by dillontkh — CHANGES_REQUESTED** [review]
Minor fix required before approval.

**Review by dillontkh — APPROVED** [review]
LGTM!

**Pull request merged**
